src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -62,20 +66,29 @@ private Database Database
                         try
                         {
                             // Get server object from connection
-                            SqlConnection sqlConn = new SqlConnection(this.serverConnection.ConnectionString);                    
+                            SqlConnection sqlConn = new SqlConnection(this.serverConnection.ConnectionString);
                             sqlConn.Open();
                             ServerConnection peekConnection = new ServerConnection(sqlConn);
-                            Server server = new Server(peekConnection);                        
-                            this.database = new Database(server, peekConnection.DatabaseName);                        
+                            Server server = new Server(peekConnection);
+                            this.database = new Database(server, peekConnection.DatabaseName);
+                        }
+                        catch (ConnectionFailureException cfe)
+                        {
+                            Logger.Write(LogLevel.Error, "Exception at PeekDefinition Database.get() : " + cfe.Message);
+                            this.error = true;
+                            this.errorMessage = connectionInfo.IsAzure ? SR.PeekDefinitionAzureError(cfe.Message) : SR.PeekDefinitionError(cfe.Message); ;
```

Comment:
You never validate connectionInfo is not null. Please check here since it's possible (in the tests there are multiple places you don't define it, in which case this would throw null ref exception)